pallets/parachain-staking/src/lib.rs

```diff
@@ -1090,10 +1091,17 @@ pub mod pallet {
 			Ok(().into())
 		}
 		/// Leave the set of nominators and, by implication, revoke all ongoing nominations
-		#[pallet::weight(T::WeightInfo::leave_nominators(T::MaxCollatorsPerNominator::get()))]
-		pub fn leave_nominators(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
+		#[pallet::weight(T::WeightInfo::leave_nominators(*nomination_count))]
+		pub fn leave_nominators(
+			origin: OriginFor<T>,
+			nomination_count: u32,
+		) -> DispatchResultWithPostInfo {
 			let acc = ensure_signed(origin)?;
 			let nominator = <NominatorState<T>>::get(&acc).ok_or(Error::<T>::NominatorDNE)?;
+			ensure!(
+				(nominator.nominations.0.len() as u32) <= nomination_count,
```

Comment:
I was thinking that as long as the hint is greater than the weight we expect, we shouldn't throw an error but this would allow someone to overpay. @notlesh what do you think

Reply:
I followed this rule for all weight hint runtime checks